FirebaseAppDistribution/Sources/FIRAppDistributionAuthPersistence.m

```diff
@@ -16,77 +16,99 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+NSString *const kFIRAppDistributionKeychainErrorDomain = @"com.firebase.app_distribution.internal";
+
 @implementation FIRAppDistributionAuthPersistence
 
-+ (BOOL)clearAuthState {
-    NSMutableDictionary *keychainQuery = [self getKeyChainQuery];
-    OSStatus status = SecItemDelete((CFDictionaryRef)keychainQuery);
-    
-    if (status != errSecSuccess && status != errSecItemNotFound) {
-        NSLog(@"AUTH ERROR. Cant delete auth state in keychain");
-    } else {
-        NSLog(@"AUTH SUCCESS! deleted auth state in the keychain");
-    }
-    return errSecSuccess || status == errSecItemNotFound;
++ (BOOL)clearAuthState:(NSError **_Nullable)error {
+  NSMutableDictionary *keychainQuery = [self getKeyChainQuery];
+  OSStatus status = SecItemDelete((CFDictionaryRef)keychainQuery);
+
+  if (status != errSecSuccess && status != errSecItemNotFound && error) {
+    NSString *desc = NSLocalizedString(
+        @"Failed to clear auth state from keychain. Tester will overwrite data on sign in.",
+        @"Error message for failure to retrieve auth state from keychain");
+    NSDictionary *userInfo = @{NSLocalizedDescriptionKey : desc};
+    *error = [NSError errorWithDomain:kFIRAppDistributionKeychainErrorDomain
+                                 code:FIRAppDistributionErrorTokenDeletionFailure
+                             userInfo:userInfo];
+  }
+  return errSecSuccess || status == errSecItemNotFound;
```

Comment:
If statement and return values do not line up (isn't `errSecSuccess` a constant?).

There's 2 ways to address this:

1) For these kindof coding patterns, it helps to always return a constant `YES` or `NO`. That way you only need the check in one place. With this pattern you also need to check the `if (error)` elsewhere, and I would recommend making a function to do this everywhere in your codebase, instead of building it into the logic. This requires discipline to always remember to use that function to set `**errors` (eg: https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseInstallations/Source/Library/Errors/FIRInstallationsErrorUtil.m#L23)
2) The other way is to just make a variable, and use it in the if statement and return value. To me this requires more discipline than 1) 

For here' we'd `return NO` in the error

Reply:
For 1), you would audit your code and never write `if (error)`, unless you were checking an `NSError *` (single * pointer)

Reply:
This advice applies to all the methods in this file that take an `NSError **`

Reply:
Added an error handling method to this class to leverage in all of the places that we are populating this error and restructured the code a bit to make it so we return immediately.